### DISCUSSION TITLE
contents().unwrap() seems not working properly

### DISCUSSION THREAD

**Discussion by chienhsiang-hung** [jquery/jquery#5084] [Q&A] [answered]
I'm trying to import HTML from another HTML with something like this:
```javascript
    // load include HTML components
    var includes = $('[data-include]');
    $.each(includes, function(){
        
        var file = 'src/html/' + $(this).data('include') + '.html';

        $(this).load(file);
    });
```
And remove the container dummy div using:
```javascript
    // reload from DOM and unwrap the div
    $('[data-include]').contents().unwrap();
```
The container div HTML like:
```html
<div data-include="navbar"></div>
```
Though it just won't work. After deployed, the container div stays still, no matter I use `.contents().unwrap()` or `replaceWith()`.
Could anyone help me to correct it? I really appreciate any help you can provide.

The [JS](https://github.com/chienhsiang-hung/javascript-language-switcher/blob/main/src/js/include.js) and [HTML](https://github.com/chienhsiang-hung/javascript-language-switcher/blob/main/index.html).

**Comment by ThakurAnkur** [comment, marked as answer]
Your code `$('[data-include]').contents().unwrap();` gets called before actual code is getting loaded since load calls are async.

As `data-include` is empty div, it doesn't make sense for jquery to empty it again.

I would suggest to move your unwrap code into load call function like below
` 
$(this).load(file, function () {
            $(this).contents().unwrap();
        });
`

**Reply by chienhsiang-hung** [reply]
> I would suggest to move your unwrap code into load call function like below

It worked after I put `$(this).contents().unwrap();` in the `load` method as you demonstrated. Thank you so much. Have a nice day.

> Your code $('[data-include]').contents().unwrap(); gets called before actual code is getting loaded since load calls are async.

Thank you for your kind explanation of my noob question. I will check the way it **async runs the code** later to strengthen my knowledge in related topics.
